azure/scope/clients.go

```diff
@@ -169,3 +205,151 @@ func setValue(settings auth.EnvironmentSettings, key string) {
 		settings.Values[key] = v
 	}
 }
+
+func (c *AzureClients) reconcileASOSecret(ctx context.Context, asoClientSecret *ASOSecretClient) error {
+	newASOSecret := &corev1.Secret{
+		Data: map[string][]byte{
+			"AZURE_SUBSCRIPTION_ID": []byte(asoClientSecret.SubscriptionID),
+		},
+	}
+
+	var owner metav1.OwnerReference
+
+	if asoClientSecret.AzureClusterIdentity != nil {
+		newASOSecret.ObjectMeta = metav1.ObjectMeta{
+			Name:      aso.GetASOSecretName(asoClientSecret.AzureClusterIdentity.GetName(), asoClientSecret.SubscriptionID),
+			Namespace: asoClientSecret.AzureClusterIdentity.GetNamespace(),
+		}
+
+		newASOSecret.Data["AZURE_TENANT_ID"] = []byte(asoClientSecret.AzureClusterIdentity.Spec.TenantID)
```

Comment:
please use constants for all the environment variable names (they are already available in the imported `auth` package), eg. `auth.TenantID`

Reply:
While I can re-use the existing constants, keep in mind that they are ASO related and some of the ASO secret data key names are slightly different compared with CAPZ ones(e.g `AZURE_CERTIFICATE_PASSWORD` in CAPZ vs `AZURE_CLIENT_CERTIFICATE_PASSWORD` in ASO or `AZURE_CERTIFICATE_PATH` in CAPZ and `AZURE_CLIENT_CERTIFICATE` in ASO).

I would prefer to use the ASO constants if they're getting exported(same as in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3698#discussion_r1267394407)